src/Parallel/Algorithm.hpp

```diff
@@ -1121,6 +1154,15 @@ bool AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>::
   }
   return true;
 }
+
+template <typename ParallelComponent, typename... PhaseDepActionListsPack>
+void AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>::
+    initiate_shutdown(const std::exception& exception) {
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg)
+  CkError("\nException caught in Algorithm:\n%s\n\nShutting down...\n",
+          exception.what());
+  sys::abort("");
```

Comment:
Q: Can you just delete this function (meaning `Algorithm.hpp` does not change in this PR, except for old line 746) and have the uncaught exception propagate to the `std::set_terminate` handler set up in `setup_error_handling`? If my understanding is correct, that would have the same effect as the current code (print error and abort), but it would require no error-handling code in Algorithm.hpp.

Reply:
no the whole point is in the future to try to do something here that will shutdown in a more controlled manner if possible, allowing existing output to finish as well as potentially doing additional output

Reply:
Since no special error handling is added in this PR, maybe don't make any changes to Algorithm.hpp yet before we know what will work?

Reply:
I think the Changes Larry is making are good here